core/src/test/java/de/danoeh/antennapod/core/service/playback/RepeatTapActionHandlerTest.java

```diff
@@ -0,0 +1,186 @@
+        verify(doubleTapAction).run();
+        verify(tripleTapAction, never()).run();
+    }
+
+    @Test
+    public void testEventSingleKeyTriplePress() {
+        handler.event(KEY_CODE, actions);
+        handler.event(KEY_CODE, actions);
+        handler.event(KEY_CODE, actions);
+
+        verifySchedulerInteractions(scheduler, tasks, atMost(3));
+        runAll(tasks);
+
+        verify(normalAction, never()).run();
+        verify(doubleTapAction, never()).run();
+        verify(tripleTapAction).run();
+    }
+
+    @Test
+    public void testEventSingleKeyQuadruplePress() {
+        handler.event(KEY_CODE, actions);
+        handler.event(KEY_CODE, actions);
+        handler.event(KEY_CODE, actions);
+        handler.event(KEY_CODE, actions);
+
+        verifySchedulerInteractions(scheduler, tasks, atMost(3));
+        runAll(tasks);
+
+        verify(normalAction, never()).run();
+        verify(doubleTapAction, never()).run();
+        verify(tripleTapAction).run();
+    }
+
+    @Test
+    public void testEventDifferentKeysSinglePress() {
+        handler.event(KEY_CODE, actions);
+
+        Runnable otherAction = mock(Runnable.class);
+        List<Runnable> otherActions = new ArrayList<>();
+        otherActions.add(otherAction);
+        handler.event(KEY_CODE + 1, otherActions);
+
+        verifySchedulerInteractions(scheduler, tasks, atMost(2));
+        runAll(tasks);
+
+        verify(normalAction).run();
+        verify(otherAction).run();
+        verify(doubleTapAction, never()).run();
+        verify(tripleTapAction, never()).run();
+    }
+
+    @Test
+    public void testEventDifferentKeysDoublePressInterleaved() {
+        Runnable otherAction = mock(Runnable.class);
+        List<Runnable> otherActions = new ArrayList<>();
+        otherActions.add(otherAction);
+
+        handler.event(KEY_CODE, actions);
+        handler.event(KEY_CODE + 1, otherActions);
+        handler.event(KEY_CODE, actions);
+
+        verifySchedulerInteractions(scheduler, tasks, atMost(3));
+        runAll(tasks);
+
+        verify(normalAction, times(2)).run();
+        verify(otherAction).run();
+        verify(doubleTapAction, never()).run();
+        verify(tripleTapAction, never()).run();
+    }
+
+    private static void verifySchedulerInteractions(Handler scheduler,
+                                                    ArgumentCaptor<Runnable> tasks) {
+        verifySchedulerInteractions(scheduler, tasks, times(1));
+    }
+
+    private static void verifySchedulerInteractions(Handler scheduler,
+                                                    ArgumentCaptor<Runnable> tasks,
+                                                    VerificationMode mode) {
+        verify(scheduler, mode).postDelayed(
+                tasks.capture(),
+                eq(RepeatTapActionHandler.DELAY_MS));
+    }
+
+    private static void runAll(ArgumentCaptor<Runnable> tasks) {
+        for(Runnable r : tasks.getAllValues()) {
```

Comment:
Another thing I noticed: If you add new code or touch existing lines, please format them according to the Google code style guidelines. You can enable Checkstyle 8.12 and load the file config/checkstyle/best-practice.xml into Intellij/AS to get the AntennaPod style (based on Google style, just a bit less enforcing in a few aspects). 

Please only update the code style for lines that you touch (eg lines that appear in diff).